pkg/execution/state/redis_state/cluster.go

```diff
@@ -0,0 +1,107 @@
+package redis_state
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/hex"
+	"github.com/redis/rueidis"
+	"runtime"
+	"time"
+)
+
+type RetriableClient interface {
```

Comment:
For Redis clustering, we need to add retry handling to our Redis client, as resizing and scaling operations may cause transient `CLUSTERDOWN` errors, which are not handled by Rueidis.